src/Adapter/Product/Update/RelatedProductsUpdater.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShopException;
+use Product;
+
+/**
+ * Provides methods to update related products (a.k.a accessories)
+ */
+class RelatedProductsUpdater
+{
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @param ProductRepository $productRepository
+     */
+    public function __construct(
+        ProductRepository $productRepository
+    ) {
+        $this->productRepository = $productRepository;
+    }
+
+    /**
+     * @param Product $product
+     * @param ProductId[] $relatedProductIds
+     */
+    public function set(Product $product, array $relatedProductIds): void
+    {
+        try {
+            if (empty($relatedProductIds)) {
+                $product->deleteAccessories();
```

Comment:
This is a doublon, you re-call `$product->deleteAccessories();` a few lines below, this could be simplified by doing:
- first check products exist
- then always perform deleteAccessories
- finally `if (!empty($ids)) { $product->changeAccessories($ids); }`

Reply:
Thats the point. When array is empty I don't need to check existence, I just delete all related products and don't go further.
Btw I just thought I might need to implement database transactions here in case changeAccessories() goes wrong after deletion. wdyt ? :thinking:

Reply:
Ideally a transaction would be nice indeed